optaplanner-examples/src/main/java/org/optaplanner/examples/pas/solver/score/PatientAdmissionMoveConstraintProvider.java

```diff
@@ -0,0 +1,182 @@
+                        && bd.getBed() != null)
+                .join(BedDesignation.class,
+                        equal(BedDesignation::getRoom),
+                        lessThan(BedDesignation::getId),
+                        filtering((left, right) -> right.getRoomGenderLimitation() == GenderLimitation.SAME_GENDER
+                                && left.getPatient().getGender() != right.getPatient().getGender()
+                                && left.getAdmissionPart().calculateSameNightCount(right.getAdmissionPart()) > 0))
+                .penalize("differentGenderInSameGenderRoomInSameNight", HardMediumSoftScore.ofHard(1000),
+                        (left, right) -> left.getAdmissionPart().calculateSameNightCount(right.getAdmissionPart()));
+    }
+
+    public Constraint departmentMinimumAgeConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(Department.class)
+                .filter(d -> d.getMinimumAge() != null)
+                .join(BedDesignation.class,
+                        equal(Function.identity(), BedDesignation::getDepartment),
+                        greaterThan(Department::getMinimumAge, BedDesignation::getPatientAge))
+                .penalize("departmentMinimumAge", HardMediumSoftScore.ofHard(100),
+                        (d, bd) -> bd.getAdmissionPartNightCount());
+    }
+
+    public Constraint departmentMaximumAgeConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(Department.class)
+                .filter(d -> d.getMaximumAge() != null)
+                .join(BedDesignation.class,
+                        equal(Function.identity(), BedDesignation::getDepartment),
+                        lessThan(Department::getMaximumAge, BedDesignation::getPatientAge))
+                .penalize("departmentMaximumAge", HardMediumSoftScore.ofHard(100),
+                        (d, bd) -> bd.getAdmissionPartNightCount());
+    }
+
+    public Constraint requiredPatientEquipmentConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(RequiredPatientEquipment.class)
+                .join(BedDesignation.class,
+                        equal(RequiredPatientEquipment::getPatient, BedDesignation::getPatient),
+                        filtering((rpe, bd) -> bd.getBed() != null))
+                .ifNotExists(RoomEquipment.class,
+                        equal((rpe, bd) -> bd.getRoom(), RoomEquipment::getRoom),
+                        equal((rpe, bd) -> rpe.getEquipment(), RoomEquipment::getEquipment))
+                .penalize("requiredPatientEquipment", HardMediumSoftScore.ofHard(50),
+                        (rpe, bd) -> bd.getAdmissionPartNightCount());
+    }
+
+    //Medium
+    public Constraint assignEveryPatientToABedConstraint(ConstraintFactory constraintFactory) {
+        return constraintFactory.fromUnfiltered(BedDesignation.class)
```

Comment:
Since `bed` is `nullable`, `fromUnfiltered()` is no different than `from()`. You might as well use `from()`.

Reply:
Yes, leftover.

Reply:
https://github.com/kiegroup/optaplanner/pull/872/commits/1f9f2b888293c0e1f544dfc79931c17386a57b41